src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/HealthHeartbeatDefaultPayload.cs

```diff
@@ -0,0 +1,125 @@
+            {
+                switch (fieldName)
+                {
+                    case "runtimeFramework":
+                        payload.Add(fieldName, new HealthHeartbeatPropertyPayload()
+                        {
+                            IsHealthy = true,
+                            PayloadValue = this.GetRuntimeFrameworkVer()
+                        });
+                        break;
+                    case "targetFramework":
+                        payload.Add(fieldName, new HealthHeartbeatPropertyPayload()
+                        {
+                            IsHealthy = true,
+                            PayloadValue = this.GetTargetFrameworkVer()
+                        });
+                        break;
+                    case UpdatedFieldsPropertyKey:
+                        var updatedFieldItem = new HealthHeartbeatPropertyPayload()
+                        {
+                            IsHealthy = true,
+                            PayloadValue = string.Empty
+                        };
+                        updatedFieldItem.IsUpdated = false; // always set this to false
+                        payload.Add(fieldName, updatedFieldItem);
+                        break;
+                    default:
+                        payload.Add(fieldName, new HealthHeartbeatPropertyPayload()
+                        {
+                            IsHealthy = false,
+                            PayloadValue = "UNDEFINED"
+                        });
+                        break;
+                }
+            }
+
+            return payload;
+        }
+
+        private void SetEnabledProperties(IEnumerable<string> disabledFields)
+        {
+            if (disabledFields == null || disabledFields.Count() <= 0)
+            {
+                this.enabledProperties = DefaultFields.ToList();
+            }
+            else
+            {
+                this.enabledProperties = new List<string>();
+                foreach (string fieldName in DefaultFields)
+                {
+                    if (!disabledFields.Contains(fieldName, StringComparer.OrdinalIgnoreCase))
+                    {
+                        this.enabledProperties.Add(fieldName);
+                    }
+                }
+            }
+        }
+
+        private string GetTargetFrameworkVer()
+        {
+#if NET45
+            return "net45";
+#elif NET46
+            return "net46";
+#elif NETSTANDARD1_3
+            return "netstandard1.3";
+#else 
+            return "undefined";
+#endif
+        }
+
+        private string GetRuntimeFrameworkVer()
+        {
+            // taken from https://github.com/Azure/azure-sdk-for-net/blob/f097add680f37908995fa2fbf6b7c73f11652ec7/src/SdkCommon/ClientRuntime/ClientRuntime/ServiceClient.cs#L214
+            try
+            {
+                Assembly assembly = typeof(Object).GetTypeInfo().Assembly;
+
+                AssemblyFileVersionAttribute objectAssemblyFileVer =
```

Comment:
there is a know case with .NET Native when there will be more than one `AssemblyFileVersionAttribute` defined. You can find an issue on GitHub if you want. Please change it to "FirstOrDefault" so it will not crash with exception saying that more than one attribute was defined

Reply:
Done, thanks for the tip.